idx_px: 96
idx_py: 100
idx_pz: 5
